sklearn/kernel_approximation.py

```diff
@@ -494,13 +494,19 @@ def fit(self, X, y=None):
             Returns the instance itself.
         """
         self._validate_params()
-        X = self._validate_data(X)
+        X = self._validate_data(X, dtype=[np.float64, np.float32])
```

Comment:
As discussed in #24317, since no computation is done using X in fit, we can let any numeric dtype through.
```suggestion
        X = self._validate_data(X)
```